###Exercício 1: Teste A/B para Avaliação de Design de Site

Uma empresa de e-commerce deseja avaliar se um novo design de sua página de produto leva a uma maior taxa de cliques (CTR) em seus produtos.

Durante uma semana, eles mostraram o design antigo a 1.000 visitantes e o novo design a 1.000 outros visitantes.

O design antigo levou a 150 cliques, enquanto o novo design levou a 180 cliques. Determine se o novo design é estatisticamente significativo.

In [2]:
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
# Dados
n_A = 1000 # numero de visitantes
n_B = 1000
conversions_A = 120
conversions_B = 180

count = np.array([conversions_B, conversions_A])
nobs = np.array([n_B, n_A])

# Usando a função proportions_ztest
z_value, p_value_statsmodels = proportions_ztest(count, nobs)

z_value, p_value_statsmodels


(3.757345746510896, 0.00017172515611363064)

#### Resultado do Teste A/B para Avaliação de Design de Site:

Com base na análise realizada, os resultados indicam que o novo design da página de produto do site tem um desempenho estatisticamente significativo em comparação com o design antigo.

Ao longo de uma semana de observação, o design antigo foi apresentado a 1.000 visitantes, resultando em 150 cliques. Enquanto isso, o novo design foi exibido a outros 1.000 visitantes, registrando 180 cliques.

Ao calcular o Z-score e o p-valor, obtivemos um Z-score de 3.757 e um p-valor de 0.0001717. Dessa forma, com um alto nível de confiança estatística, rejeitamos a hipótese nula, que afirmava que não há diferença significativa entre os designs. Portanto, aceitamos a hipótese alternativa de que o novo design da página resulta em uma taxa de cliques significativamente maior.

Consequentemente, com base nos dados e na análise estatística realizada, é recomendável implementar o novo design da página de produto como a versão padrão do site, a fim de potencializar a taxa de cliques e possivelmente melhorar o desempenho geral do e-commerce.

###Exercício 2: Teste A/B para Preços
Uma empresa SaaS (Software as a Service) está considerando aumentar o preço de sua assinatura mensal de R$20 para R$25. Eles testam a nova estrutura de preços em 500 visitantes e mantêm o preço antigo para outros 500 visitantes. Com o preço antigo, 200 usuários se inscreveram, enquanto com o novo preço, 180 usuários se inscreveram. O novo preço é vantajoso para a empresa?





In [4]:
# Dados
n_A = 500 # numero de visitantes
n_B = 500
conversions_A = 200
conversions_B = 180

count = np.array([conversions_B, conversions_A])
nobs = np.array([n_B, n_A])

# Usando a função proportions_ztest
z_value, p_value_statsmodels = proportions_ztest(count, nobs)

z_value, p_value_statsmodels

(-1.3029938101426084, 0.19257687480088492)

####Resultado do Teste A/B para Avaliação de preços da assinatura mensal da empresa SaaS:

o valor Z negativo indica que a média de conversões (ou inscrições) com o novo preço é menor do que a média com o preço antigo.

um p-valor alto indica que há uma probabilidade considerável de observar uma diferença nas inscrições entre os dois grupos simplesmente por acaso, mesmo se não houver realmente uma diferença entre os preços.


###Exercício 3: Teste A/B para Avaliação de Email Marketing
Uma empresa de varejo online decidiu testar duas versões diferentes de um email promocional. O email da Versão A foi enviado para 2.000 clientes e teve uma taxa de abertura de 20%. O email da Versão B foi enviado para 2.500 clientes e teve uma taxa de abertura de 22%. Qual versão do email teve melhor desempenho?

In [5]:
# Dados
n_A = 2000 # numero de visitantes
n_B = 2500
conversions_A = 0.20
conversions_B = 0.22

count = np.array([conversions_B, conversions_A])
nobs = np.array([n_B, n_A])

# Usando a função proportions_ztest
z_value, p_value_statsmodels = proportions_ztest(count, nobs)

z_value, p_value_statsmodels

(-0.041405865879370805, 0.9669723365138204)

####Resultado do Teste A/B para avaliação de envio de email de marketing:

Como o p-valor é maior que o nível de significância comum de 0.05 (ou 5%), não temos evidências suficientes para rejeitar a hipótese nula. Portanto, não podemos concluir com confiança que uma versão do email teve um desempenho melhor em termos de taxa de abertura do que a outra.

Em resumo, os resultados do teste A/B não mostram uma diferença estatisticamente significativa entre as duas versões do email em termos de taxa de abertura. Portanto, não há evidências de que uma versão seja superior à outra.

###Exercício 4: Teste A/B para Avaliação de Tempo no Site
Uma plataforma de notícias online implementou uma nova característica que sugere artigos relacionados ao usuário. Eles querem testar se essa característica aumenta o tempo médio que um usuário passa no site. Eles implementam o novo recurso para 1.500 usuários e veem um tempo médio de permanência de 5,2 minutos, com um desvio padrão de 1,5 minutos. Para 1.700 usuários sem o novo recurso, o tempo médio de permanência foi de 5 minutos com um desvio padrão de 1,4 minutos. A nova característica fez alguma diferença

In [6]:
import numpy as np
from scipy.stats import ttest_ind

# Dados do problema
n1 = 1500
mean1 = 5.2
std_dev1 = 1.5

n2 = 1700
mean2 = 5.0
std_dev2 = 1.4

# Calculando o desvio padrão combinado (standard error)
SE = np.sqrt((std_dev1**2 / n1) + (std_dev2**2 / n2))

# Calculando o valor t
t_value = (mean1 - mean2) / SE

# Calculando o p-valor usando ttest_ind
t_statistic, p_value = ttest_ind(np.zeros(n1) + mean1, np.zeros(n2) + mean2, equal_var=False)

print("Valor t:", t_value)
print("P-valor:", p_value)


Valor t: 3.882989225005368
P-valor: 0.0


/usr/local/lib/python3.10/dist-packages/scipy/stats/_axis_nan_policy.py:523: RuntimeWarning: Precision loss occurred in moment calculation due to catastrophic cancellation. This occurs when the data are nearly identical. Results may be unreliable.
  res = hypotest_fun_out(*samples, **kwds)


####Resultado do Teste A/B para avaliação de tempo no site:

Com base nos resultados, o valor t positivo e o p-valor muito baixo indicam que a diferença nos tempos médios de permanência entre os dois grupos (com e sem a nova característica) é estatisticamente significativa.

Como o p-valor é significativamente menor que o nível de significância comum de 0.05 (ou 5%), podemos rejeitar a hipótese nula. Portanto, podemos concluir que a implementação da nova característica fez uma diferença no tempo médio de permanência dos usuários no site.

Os resultados indicam que a implementação da nova característica que sugere artigos relacionados aumentou significativamente o tempo médio que um usuário passa no site.

###Exercício 5: Teste A/B para Avaliação de Taxa de Rejeição
Uma empresa de software quer testar se uma nova introdução tutorial para seu software reduz a taxa de rejeição dos novos usuários. Dos 1.200 novos usuários que viram o tutorial, 400 abandonaram o software no primeiro uso. Dos 1.100 usuários que não viram o tutorial, 500 abandonaram o software no primeiro uso. O tutorial fez alguma diferença na taxa de rejeição?

In [12]:
# Dados
n_A = 1200 # viram o tutorial
n_B = 1100 #não viram o tutorial
conversions_A = 400
conversions_B = 500

count = np.array([conversions_B, conversions_A])
nobs = np.array([n_B, n_A])

# Usando a função proportions_ztest
z_value, p_value_statsmodels = proportions_ztest(count, nobs)

z_value, p_value_statsmodels

(5.949927344231642, 2.682615441474503e-09)

####Resultado do Teste A/B para avaliação da taxa de rejeição:
Como o p-valor é significativamente menor que o nível de significância comum de 0.05 (ou 5%), podemos rejeitar a hipótese nula. Portanto, podemos concluir que a introdução do novo tutorial fez uma diferença na taxa de rejeição dos novos usuários do software.

Os resultados indicam que a introdução do tutorial teve um impacto positivo na redução da taxa de rejeição dos novos usuários do software.